packages/server-wallet/src/protocols/close-channel.ts

```diff
@@ -216,6 +217,22 @@ const defundIntoLedger = (ps: ProtocolState): RequestLedgerDefunding | false =>
     assetHolderAddress: checkThat(ps.app.latest.outcome, isSimpleAllocation).assetHolderAddress,
   });
 
+/**
+ * If the channel is directly funded, then ensure none of its allocation items are other
+ * channels being funded by this channel (e.g., if it is a ledger channel). This should
+ * cause the protocol to "pause" / "freeze" until no channel depends on this channel
+ * for funding.
+ */
+const ensureAllAllocationItemsAreExternalDestinations = (ps: ProtocolState): boolean =>
+  (!isDirectlyFunded(ps.app) && !isFakeFunded(ps.app)) ||
+  ((isDirectlyFunded(ps.app) || isFakeFunded(ps.app)) &&
```

Comment:
The name of the function suggests no condition on funding. The docstring suggests the condition should be `isDirectlyFunded` . The  the actual condition seems to be either directly funded or fake funded".

I'd like us to consider here what the long term rule should be for the wallet, and why. Perhaps the rule could be "do not close any channel that allocates to other channels, regardless of funding". (In that case, you should keep the name of the function but remove the conditional logic highlighted).

Reply:
Hm yeah, "do not close any channel that allocates to other channels, regardless of funding" might be better